loggers/serial_writer.go

```diff
@@ -16,22 +18,31 @@ import (
 
 // serialWriter is used to store logs in a single file
 type serialWriter struct {
-	file      *os.File
+	file      io.Closer
 	log       log.Logger
 	formatter LogFormatter
 }
 
 func (sl *serialWriter) Init(c types.Config) error {
-	fileName := c.Get("auditlog_file", "/dev/null").(string)
 	fileMode := c.Get("auditlog_file_mode", fs.FileMode(0644)).(fs.FileMode)
 	sl.formatter = c.Get("auditlog_formatter", nativeFormatter).(LogFormatter)
-	var err error
-	sl.file, err = os.OpenFile(fileName, os.O_APPEND|os.O_CREATE|os.O_WRONLY, fileMode)
-	if err != nil {
-		return err
+
+	fileName := c.Get("auditlog_file", "").(string)
+	var file io.Writer
+	if fileName != "" {
+		f, err := os.OpenFile(fileName, os.O_APPEND|os.O_CREATE|os.O_WRONLY, fileMode)
+		if err != nil {
+			return err
+		}
+		file = f
+		sl.file = f
+	} else {
+		file = io.Discard
+		// Nothing to close so just set a no-op
+		sl.file = io.NopCloser(&bytes.Buffer{})
```

Comment:
We still don't have a mechanism to close WAF instances. It should handle closing log writters and audit writters.

Reply:
Just to confirm this is something to fix in another PR? Here we're just keeping the existing code working, i.e., to support the `Close` method on line 58